src/presence_analyzer/tests.py

```diff
@@ -354,6 +355,30 @@ def test_group_start_end_by_weekday(self):
 
         self.assertEqual(result, utils.group_start_end_by_weekday(fake_data))
 
+    def test_cache_decorator(self):
+        """
+        Test utils/cache result
+        """
+        moc = MagicMock()
+        _func = lambda _: moc()
+        cachelimit = 0.001
+
+        wrapped = utils.cache(cachelimit)(_func)
+
+        wrapped(1)
+        wrapped(2)
+        self.assertEqual(moc.call_count, 2)
+
+        wrapped(1)
+        self.assertEqual(moc.call_count, 2)
+
+        now = datetime.datetime.now()
+        with patch('datetime.datetime') as dt:
```

Comment:
Your mock is not working, if you used IPDB, `s`tepped into `with patch` and checked `datetime.now`, you would see built-in python function, not mock.
`with patch('presence_analyzer.utils.datetime') as dt:`
Then you can remove `reload(utils)`